### DISCUSSION TITLE
Fit outside data using precomputed space.

### DISCUSSION THREAD

**Discussion by JacobCWBillings** [lmcinnes/umap#650] [Q&A] [answered]
Hello,

I'm wondering if new data can be fit to a model trained using a precomputed distance metric? The current code-base throws a warning about this when using precomputed metrics. But intuitively, one needs only to supply precomputed distances between each out-of-sample node and the set of within-sample nodes to fit new data. 

What else is going on?

Many thanks, in advance, 

-Jacob

**Comment by JacobCWBillings** [comment, marked as answer]
I see a note about this in the PR archives: https://github.com/lmcinnes/umap/pull/639#issue-607634060

Seems like the intuition was a good one +1

**Comment by lmcinnes** [comment]
Yes, it is now possible if you pull the latest version on github. That feature is newer than the latest release onto PyPI however, so if you are installing from pip or conda it won't be available.

**Comment by JacobCWBillings** [comment]
Perfect. 

I'm just glad (for posterity) that there isn't something in the _mere calculation_ of distance functions that UMAP (and other algorithms?) needs. Rather, an empirically predefined distance metric space is sufficient to compute dimensionality reduction of outside points.

**Reply by JacobCWBillings** [reply]
Many, many thanks for the contribution of UMAP, Leland!
